### ISSUE TITLE
Introduce tag system

### ISSUE THREAD

**Issue by stevewhitmore** [stevewhitmore/archium#1]
Pages should be tagged for easier organization and accessibility. Pages will
- be searchable by tag
- be allowed to have more than 1 tag
- be allowed to choose from predefined tags
- be able to create new tags
- be addable/removable in a separate section
